src/WebJobs.Script/Host/BlobLeaseManager.cs

```diff
@@ -78,24 +81,24 @@ private set
 
         private void OnHasLeaseChanged() => HasLeaseChanged?.Invoke(this, EventArgs.Empty);
 
-        public static async Task<BlobLeaseManager> CreateAsync(string accountConnectionString, TimeSpan leaseTimeout, string hostId, string instanceId,
-            TraceWriter traceWriter, ILoggerFactory loggerFactory)
+        public static BlobLeaseManager Create(
+            IDistributedLockManager lockManager,
+            TimeSpan leaseTimeout,
+            string hostId,
+            string instanceId,
+            TraceWriter traceWriter,
+            ILoggerFactory loggerFactory,
+            TimeSpan? renewalInterval = null)
         {
             if (leaseTimeout.TotalSeconds < 15 || leaseTimeout.TotalSeconds > 60)
             {
                 throw new ArgumentOutOfRangeException(nameof(leaseTimeout), $"The {nameof(leaseTimeout)} should be between 15 and 60 seconds");
```

Comment:
Good point. Although there's only one callsite here (in ScriptHost.cs) and the value passed in is a compile-time constant. So this check is of questionable value. 

---
In reply to: [123670795](https://github.com/Azure/azure-webjobs-sdk-script/pull/1615#discussion_r123670795) [](ancestors = 123670795)